src/components/Navbar.jsx

```diff
@@ -1,35 +1,83 @@
 import React from 'react';
-import { NavLink } from 'react-router-dom';
+import { NavLink, useLocation } from 'react-router-dom';
 import { RiCommunityFill } from "react-icons/ri";
 import { FaRecycle } from "react-icons/fa";
 import { MdOutlineRestaurantMenu } from "react-icons/md";
 import { RiCoupon3Fill } from "react-icons/ri";
 
 const Navbar = () => {
+  const location = useLocation();
+
+  const shouldHighlight = (route) => {
```

Comment:
This overall looks messy, should we consider changing our route paths to make each route path starts with their corresponding route name? (e.g. `/coupons` -> `/reward/coupons`) In this way, I believe this function can be simply rewritten as:
``` js
  const shouldHighlight = (route) => {
    return location.pathname.startsWith(route);
  };
```
What do you guys think?  @liyangc2S2023 @YinzhuSU @xyang6230 @hanqidS2023 @ziyangcS2023

Reply:
Im okay with it. For login/logout, I'll do login/restaurant, login/recycle, login/customer, logout will be **logout/flowname (e.g. logout/community)**